Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp

```diff
@@ -246,24 +268,18 @@ void GStreamerCaptureDeviceManager::refreshCaptureDevices()
     if (!m_deviceMonitor) {
         m_deviceMonitor = adoptGRef(gst_device_monitor_new());
 
-        switch (deviceType()) {
-        case CaptureDevice::DeviceType::Camera: {
-            gst_device_monitor_add_filter(m_deviceMonitor.get(), "Video/Source", nullptr);
-            break;
+        auto types = deviceTypes();
+        if (types.contains(CaptureDevice::DeviceType::Camera)) {
+            auto caps = adoptGRef(gst_caps_new_empty_simple("video/x-raw"));
```

Comment:
Why are we just restricting to raw video cameras? There are not so many that can accept raw video with higher resolutions.

Reply:
Yes, encoded formats should  also work, the capturer currently decodes those, which isn't great either but is what it is.

Reply:
This was in the original patch by @philn on which this PR is based, so I assumed it was correct and kept it 😉

Reply:
Hehe :) Maybe that initial patch was done before decodebin support was added to the capturer. Can you revert this part?

Reply:
Of course, will do 👌🏼